server/service/src/invoice/validate.rs

```diff
@@ -105,3 +108,18 @@ pub fn check_invoice_exists(
         Err(error) => Err(error),
     }
 }
+
+pub fn check_invoice_does_not_exists(
+    id: &str,
+    connection: &StorageConnection,
+) -> Result<(), InvoiceAlreadyExistsError> {
+    let result = InvoiceRowRepository::new(connection).find_one_by_id(id);
+
+    if let Err(RepositoryError::NotFound) = &result {
```

Comment:
Note, can do with a refactor to return an optional when not found...

Reply:
This would be great refactor for another PR (to keep code changes [targeted](https://github.com/openmsupply/open-msupply/wiki/Code-Review-Comments---General#should-i-make-small-unrelated-fixes-in-my-pr)), i thought there was an issue for it, but can't find it (must be a subtask somewhere). Would also include other repositories, and removal of this one: 
https://github.com/openmsupply/open-msupply/blob/79c8572aff0ba7d2c06316e6d423611091fedcd0/server/service/src/sync/test/mod.rs#L170-L177